tests/relay/test_publish.py

```diff
@@ -1,34 +1,231 @@
+                try:
+                    self.check_published_message_reaches_peer(message)
+                    success_timestamps.append(timestamp)
+                except Exception as e:
+                    pass
+        assert not success_timestamps, f"Invalid Timestamps that didn't failed: {success_timestamps}"
+
+    def test_publish_with_no_timestamp(self):
+        message = {"payload": to_base64(self.test_payload), "contentTopic": self.test_content_topic}
+        self.check_published_message_reaches_peer(message)
+
+    def test_publish_with_valid_version(self):
+        self.check_published_message_reaches_peer(self.create_message(version=10))
+
+    def test_publish_with_invalid_version(self):
+        try:
+            self.check_published_message_reaches_peer(self.create_message(version=2.1))
+            raise AssertionError("Publish with invalid version worked!!!")
+        except Exception as ex:
+            assert "Bad Request" in str(ex)
+
+    def test_publish_with_valid_meta(self):
+        self.check_published_message_reaches_peer(self.create_message(meta=to_base64(self.test_payload)))
+
+    def test_publish_with_invalid_meta(self):
+        try:
+            self.check_published_message_reaches_peer(self.create_message(meta=self.test_payload))
+            raise AssertionError("Publish with invalid meta worked!!!")
+        except Exception as ex:
+            assert "Bad Request" in str(ex)
+
+    def test_publish_with_ephemeral(self):
+        failed_ephemeral = []
+        for ephemeral in [True, False]:
+            logger.debug("Running test with Ephemeral %s", ephemeral)
+            try:
+                self.check_published_message_reaches_peer(self.create_message(ephemeral=ephemeral))
+            except Exception as e:
+                logger.error("Massage with Ephemeral %s failed: %s", ephemeral, str(e))
+                failed_ephemeral.append(ephemeral)
+        assert not failed_ephemeral, f"Ephemeral that failed: {failed_ephemeral}"
+
+    def test_publish_with_rate_limit_proof(self):
+        rate_limit_proof = {
+            "proof": to_base64("proofData"),
+            "epoch": to_base64("epochData"),
+            "nullifier": to_base64("nullifierData"),
+        }
+        self.check_published_message_reaches_peer(self.create_message(rateLimitProof=rate_limit_proof))
+
+    def test_publish_with_extra_field(self):
+        self.check_published_message_reaches_peer(self.create_message(extraField="extraValue"))
+
+    def test_publish_and_retrieve_duplicate_message(self):
+        message = self.create_message()
+        self.check_published_message_reaches_peer(message)
+        try:
+            self.check_published_message_reaches_peer(message)
+            raise AssertionError("Duplicate message was retrieved twice")
+        except Exception as ex:
+            assert "Peer node couldn't find any messages" in str(ex)
+
+    def test_publish_after_node_pauses(self):
```

Comment:
Wouldn't this test imply a `check_published_message_reaches_peer` in between a `pause` and `unpause`?

Reply:
indeed the name is off, updating to test_publish_after_node_pauses_and_unpauses
having a `check_published_message_reaches_peer `in between a `pause `and `unpause `not sure if it makes sense because the connection is down when the node is in pause state

Reply:
Yes, I also felt that way at first when I implemented something similar in `nwaku`, but after actually running the test I discovered a behaviour:
* Relay can't receive stuff while paused
* Filter CAN receive stuff while paused

Although maybe it makes sense to just have that kind of test in the Filter suite.

Reply:
I actually tried it and it worked!
Node1 - started
Node2 - paused
Node1 - publishes message
Node2 - unpasues
Node2 - gets messages 
=> Message is received

I added a new test for this scenario. Thanks for the suggestion!

Reply:
No problem, it's my pleasure to give you more work 😂